src/Nethermind/Nethermind.Evm/BlockHashInState/BlockHashInStateHandler.cs

```diff
@@ -27,9 +27,14 @@ public static void AddParentBlockHashToState(BlockHeader blockHeader, IReleaseSp
         stateProvider.Set(blockHashStoreCell, parentBlockHash.Bytes.WithoutLeadingZeros().ToArray());
     }
 
-    public static Hash256? GetBlockHashFromState(long blockNumber, IReleaseSpec spec, IWorldState stateProvider)
+    public static Hash256? GetBlockHashFromState(long currentBlockNumber, long requiredBlockNumber, IReleaseSpec spec, IWorldState stateProvider)
     {
-        var blockIndex = new UInt256((ulong)(blockNumber % Eip2935Constants.RingBufferSize));
+        if (requiredBlockNumber >= currentBlockNumber || requiredBlockNumber <
```

Comment:
why not to do the check in the same way like in EIP
  if arg >= block.number or (arg + HISTORY_SERVE_WINDOW) < block.number
  
  return null? not 0?

Reply:
its just a additional check for the condition where  we are doing this on, lets say blockNumber 7000 adn we want hash of block 0
if we dont add min, we wont get the required result - it will return null

the null -> 0 thing is already handled in the opCode
i can modify this and currect blockhash implemtation to return 0 instead of null if you want.